modyn/supervisor/internal/triggers/model_downloader/model_downloader.py

```diff
@@ -0,0 +1,106 @@
+import io
+import json
+import logging
+import pathlib
+from typing import Optional
+
+import grpc
+import torch
+from modyn.common.ftp import download_trained_model
+from modyn.metadata_database.metadata_database_connection import MetadataDatabaseConnection
+from modyn.metadata_database.models import TrainedModel
+
+# pylint: disable-next=no-name-in-module
+from modyn.model_storage.internal.grpc.generated.model_storage_pb2 import FetchModelRequest, FetchModelResponse
+from modyn.model_storage.internal.grpc.generated.model_storage_pb2_grpc import ModelStorageStub
+from modyn.models.coreset_methods_support import CoresetSupportingModule
+from modyn.utils import dynamic_module_import
+from modyn.utils.utils import grpc_connection_established
+
+logger = logging.getLogger(__name__)
+
+
+# TODO(366) Unify similar code in trainer_server and evaluator. Create common.utils.ModelDownloader
+class ModelDownloader:
```

Comment:
This is a ModelDownloader + a class representing the downloaded model. To make the object oriented abstraction a bit cleaner, could we instead just `return` a model at the end? From my understanding, the model downloader should download + instantiate, but not _own_ the model. That would be cleaner. Thank you!

Reply:
The current model class does not store additional info as attributes. E.g. amp, device, device type. I have this model downloader class as a unified interface to access such additional info required to run the forward pass. A better way would probably be to implement an abstract model class that contains those additional info. I think it would better be dealt with in the ModelDownloader Issue #433. What do you think?